app/auth/register.tsx

```diff
@@ -0,0 +1,300 @@
+import { SIGN_UP_MUTATION } from '@/graphql/mutations/register.mutation';
+import { RegisterSchema } from '@/validations/register.schema';
+import { ApolloError, useMutation } from '@apollo/client';
+import { Ionicons } from '@expo/vector-icons';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { Picker } from '@react-native-picker/picker';
+import Checkbox from 'expo-checkbox';
+import { useLocalSearchParams, useRouter } from 'expo-router';
+import { useFormik } from 'formik';
+import { jwtDecode } from 'jwt-decode';
+import { useEffect, useState } from 'react';
+import {
+  ActivityIndicator,
+  Alert,
+  TextInput,
+  ToastAndroid,
+  TouchableOpacity,
+  useColorScheme,
+} from 'react-native';
+import { SvgXml } from 'react-native-svg';
+
+type FormValues = {
+  firstName: string;
+  lastName: string;
+  password: string;
+  gender: string;
+  dob: string;
+};
+
+type TokenPayload = {
+  name: string;
+  email: string;
+  exp: number;
+};
+
+type RegisterResponse = {
+  createUser: {
+    token: string;
+  };
+};
+
+export default function RegisterForm() {
+  const router = useRouter();
+  const params = useLocalSearchParams();
+  const colorScheme = useColorScheme();
+
+  let { token } = params as { token: string };
+
+  token = Array.isArray(token) ? token.join('.') : token?.replace(/\*/g, '.');
+
+  const [loading, setLoading] = useState(false);
+  const [TCAccepted, setTCAccepted] = useState(false);
+  const [showPassword, setShowPassword] = useState(false);
+  const [email, setEmail] = useState<string | null>(null);
+  const [orgName, setOrgName] = useState<string | null>(null);
+  const [SignupUser] = useMutation<RegisterResponse>(SIGN_UP_MUTATION);
+
+  useEffect(() => {
+    const checkAuthStatus = async () => {
+      const authToken = await AsyncStorage.getItem('org_token');
+      if (authToken) {
+        try {
+          const parsedToken = jwtDecode<TokenPayload>(authToken);
+          if (parsedToken.exp < Date.now()) {
+            await AsyncStorage.removeItem('org_token');
+          } else {
+            router.push('/dashboard/trainee');
+          }
+        } catch (err) {}
+      }
+    };
+
+    checkAuthStatus();
+  }, [router]);
+
+  const formik = useFormik<FormValues>({
+    initialValues: {} as FormValues,
+    onSubmit: async (values: FormValues) => {
+      setLoading(true);
+      try {
+        const { data, errors } = await SignupUser({
+          variables: {
+            firstName: values.firstName,
+            lastName: values.lastName,
+            dateOfBirth: values.dob,
+            gender: values.gender,
+            email: email,
+            password: values.password,
+            orgToken: token,
+          },
+        });
+
+        if (data) {
+          ToastAndroid.show('Successfully registered', ToastAndroid.LONG);
+          await AsyncStorage.setItem('org_token', data.createUser.token);
+          router.push('/dashboard/trainee');
```

Comment:
After registering, why are you redirecting on the dashboard

Reply:
Changed this to redirect to login